### PR TITLE
feat(cocoapods): Add custom error message

### PR DESCRIPTION
**Pull request by wkl3nk**
Add a custom error message for the case that a build target cannot be found in the resolved versions from the lockfile. Give context information what resolved versions were found.

This improves the current situation where just an
IllegalStateException is thrown without any further context.

### CODE REVIEW

File: plugins/package-managers/cocoapods/src/main/kotlin/CocoaPods.kt [hunk 1]
```diff
@@ -252,7 +252,9 @@ private fun parseLockfile(podfileLock: File): LockfileData {
         // The version written to the lockfile matches the version specified in the project's ".podspec" file at the
         // given revision, so the same version might be used in different revisions. To still get a unique identifier,
         // append the revision to the version.
-        val versionFromPodspec = checkNotNull(resolvedVersions[name])
+        val versionFromPodspec = checkNotNull(resolvedVersions[name]) {
+            "No version found for '$name' in resolved versions $resolvedVersions."
+        }
```
Comment (fviernau) [inline plugins/package-managers/cocoapods/src/main/kotlin/CocoaPods.kt:257]: I believe we need a new line after `}`.

Maybe rephrase to: "Could not find the resolved version for '$name' in the podspec file." ?

(e.g. "in resolved version" seems to refer to the variable name of the map, which is maybe not known by the reader of the error)

Comment (wkl3nk) [reply plugins/package-managers/cocoapods/src/main/kotlin/CocoaPods.kt:257]: Agreed.

**Review by sschuberth — APPROVED** [review]

**Comment by sschuberth** [comment]
Merging despite `funTest-docker` being stuck.
